homeassistant/components/tradfri/sensor.py

```diff
@@ -27,100 +40,91 @@ async def async_setup_entry(
     coordinator_data = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
     api = coordinator_data[KEY_API]
 
-    entities: list[TradfriBaseEntity] = [
-        TradfriBatterySensor(
-            device_coordinator,
-            api,
-            gateway_id,
-        )
-        for device_coordinator in coordinator_data[COORDINATOR_LIST]
+    entity_list: list[TradfriSensor] = []
+
+    for device_coordinator in coordinator_data[COORDINATOR_LIST]:
         if (
             not device_coordinator.device.has_light_control
             and not device_coordinator.device.has_socket_control
             and not device_coordinator.device.has_signal_repeater_control
             and not device_coordinator.device.has_air_purifier_control
-        )
-    ]
-
-    entities.extend(
-        [
-            AirQualitySensor(
-                device_coordinator,
-                api,
-                gateway_id,
+        ):
+            entity_list.append(
+                TradfriSensor(
+                    device_coordinator,
+                    api,
+                    gateway_id,
+                    description=TradfriSensorEntityDescription(
```

Comment:
Can we define the descriptions as constants in a container and look up the correct description based on what device control the device has? Ie, define some id for each description that can be resolved using the device control support.

Reply:
I tried to implement this but failed to see how. Let's see if I can figure it out while writing down what I've tried so far.

All devices share the same properties but will have one of the `has_nnn_control` flags set to true. All devices also have `_control` properties, although only one of them will contain data per device. I played around with an idea of adding some sort of filter while looping through devices and device descriptions, but ended up with a lot of code so that didn't quite feel like a good idea.

Maybe we could use this as a key?
https://github.com/home-assistant-libs/pytradfri/blob/4b978321d439f48dbf88f9de19f032b23b3e0807/pytradfri/device/__init__.py#L114

Reply:
For now we could skip the I'd and just move the descriptions to be constants at the module level. Then just pass the corresponding constant here instead of defining them here.

Reply:
Noted, I've moved them.